graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java

```diff
@@ -433,17 +429,39 @@ public Node(Reports.Trace.Node.Builder builder) {
 
     public static class Options {
         private final boolean debuggingEnabled;
+        private final Predicate<Object> contextSignalsToTrace;
 
-        public Options(boolean debuggingEnabled) {
+        public Options(boolean debuggingEnabled, Predicate<Object> contextSignalsToTrace) {
             this.debuggingEnabled = debuggingEnabled;
+            this.contextSignalsToTrace = contextSignalsToTrace;
+        }
+
+        public Options(boolean debuggingEnabled) {
+            this(debuggingEnabled,
+                // Default implementation:
+                // If context implements our HTTPRequestHeaders interface, we should only be active
+                // if the special HTTP header has the special value. If the header isn't provided or has
+                // a different value, return false - which is interpreted as meaning "don't instrument".
+                // (If context doesn't implement HTTPRequestHeaders, always instrument.)
+                 (context) -> {
+                     if (context instanceof HTTPRequestHeaders) {
+                        String header = ((HTTPRequestHeaders) context).getHTTPRequestHeader(HEADER_NAME);
+                        return HEADER_VALUE.equals(header);
+                    }
+                    return true;
```

Comment:
Original implementation intact as the default, just slightly altered due to new way the question is being asked.